cdap-ui/app/cdap/components/DataPrep/ColumnActionsDropdown/index.js

```diff
@@ -40,39 +44,73 @@ export default class ColumnActionsDropdown extends Component {
 
     this.state = {
       dropdownOpen: false,
-      open: null
+      open: null,
+      selectedHeaders: DataPrepStore.getState().dataprep.selectedHeaders
     };
 
     this.toggleDropdown = this.toggleDropdown.bind(this);
 
+
+    /*
+      colSize attribute refers to the number of selected columns a directive needs to work
+    e.g. MergeColumnDirective and SwapColumnDirective work when and only when the number of
+    selected columns is 2.
+      The possible values are:
+    - 0: Works with any number of selected columns.
+    - 1: Works only when there is 1 selected column.
+    - 2: Works only when there are 2 selected columns.
+
+    */
     this.directives = [
       {
         id: shortid.generate(),
-        tag: FilterDirective
+        tag: FilterDirective,
+        colSize: 1
```

Comment:
May be `requiredColCount`? `colSize` signifies more on the size of the column whereas here we want minimum number of columns to be selected to apply the directive.